src/compiler/crystal/compiler.cr

```diff
@@ -243,8 +262,8 @@ module Crystal
 
     private def bc_flags_changed?(output_dir)
       bc_flags_changed = true
-      current_bc_flags = "#{@codegen_target}|#{@mcpu}|#{@mattr}|#{@release}|#{@link_flags}|#{@mcmodel}"
-      bc_flags_filename = "#{output_dir}/bc_flags"
+      current_bc_flags = "#{@codegen_target}|#{@mcpu}|#{@mattr}|#{@link_flags}|#{@mcmodel}"
+      bc_flags_filename = "#{output_dir}/bc_flags#{optimization_mode_suffix}"
```

Comment:
**question:** Putting the optimization level in the filename seems like a great idea. It changes from the current behaviour where it's only written in the file contents.
I'm wondering about the effects of this. 
I suppose it means the caches for different optimization modes won't override each other.
Does it mean different caches stay around? But what about the actual data files in `output_dir`?

Reply:
Before this PR, all objects files was mixed, but release used only 2 files: _main.o and _main.bc, so mixing was not so important. But in this PR added many modes, so every mode have separate compiled objects(.o and .bc) for each class.
About the actual data, still this is only cache files, its became useless, it can be removed just with `rm -rf ~/.cache/crystal`. And of course all caches for each mode would stay around. But in every day usage this cache would be the similar as before this PR, because default compile - would generate .o0 files, and release would generate just 2 files _main.o.o3 _main.bc.o3